### PR TITLE
tsp, LRO, integrate with getLroMetadata and core-experimental

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
link https://github.com/Azure/azure-sdk-for-java/pull/34525

partly handle https://github.com/Azure/autorest.java/issues/1962

main change on generated code is https://github.com/Azure/autorest.java/pull/2098/commits/7b804b945dccd6f0937565797352551e32ddb2f0

### CODE REVIEW

File: typespec-tests/pom.xml [hunk 1]
```diff
@@ -42,6 +42,11 @@
       <artifactId>azure-identity</artifactId>
       <version>1.8.2</version>
     </dependency>
+    <dependency>
+      <groupId>com.azure</groupId>
+      <artifactId>azure-core-experimental</artifactId>
+      <version>1.0.0-beta.39</version>
```
Comment (weidongxu-microsoft) [inline typespec-tests/pom.xml:48]: this version currently is only local build, from https://github.com/Azure/azure-sdk-for-java/pull/34525

File: typespec-tests/src/main/java/com/azure/lro/core/CoreAsyncClient.java [hunk 2]
```diff
@@ -111,15 +111,7 @@ public PollerFlux<BinaryData, Void> beginDelete(String name, RequestOptions requ
...
-     * {
-     *     id: String (Required)
-     *     status: String(InProgress/Succeeded/Failed/Canceled) (Required)
-     *     error: ResponseError (Optional)
-     *     result (Optional): {
-     *         name: String (Required)
-     *         resourceUri: String (Required)
-     *     }
-     * }
+     * PollResult
```
Comment (weidongxu-microsoft) [inline typespec-tests/src/main/java/com/azure/lro/core/CoreAsyncClient.java:114]: Here problem about handling external model. Tracking issue for improvement https://github.com/Azure/autorest.java/issues/2097

Comment (weidongxu-microsoft) [reply typespec-tests/src/main/java/com/azure/lro/core/CoreAsyncClient.java]: fixed https://github.com/Azure/autorest.java/pull/2098/commits/9f37bbe146cc16b89542aa6eb2a4789aeaeb7b9f

File: typespec-extension/src/code-model-builder.ts [hunk 3]
```diff
@@ -544,15 +553,94 @@ export class CodeModelBuilder {
...
+
+    const lroMetadata = getLroMetadata(this.program, operation);
+    if (
+      lroMetadata &&
+      // we know those operation with OperationStatus is from Azure.Core,
+      // which is validated that "getLroMetadata" gives correct metadata.
+      // there are known cases that on legacy LRO, "getLroMetadata" gives wrong metadata.
+      lroMetadata.statusMonitorStep?.responseModel &&
+      lroMetadata.statusMonitorStep.responseModel.name === "OperationStatus" &&
+      getNamespace(lroMetadata.statusMonitorStep.responseModel) === "Azure.Core.Foundations"
```
[the remark this answers is not in the document]
Comment (weidongxu-microsoft) [reply typespec-extension/src/code-model-builder.ts]: Moved to `isLroMetadataSupported` https://github.com/Azure/autorest.java/pull/2098/commits/e189feefe6b17d64ee854df5f5965ff5b131db6e

File: javagen/src/main/java/com/azure/autorest/util/SchemaUtil.java [hunk 4]
```diff
@@ -228,14 +228,25 @@ public static Optional<ClassType> mapExternalModel(ObjectSchema compositeType) {
...
                     }
-                    // ErrorResponse is not available
+                    // ErrorResponse is not available, but that should only be used in Exception
+
+                    // https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/models/PollResult.java
+                    if ("PollResult".equals(name)
+                            && compositeType.getLanguage().getJava() != null
+                            && compositeType.getLanguage().getJava().getNamespace() != null
+                            && compositeType.getLanguage().getJava().getNamespace().startsWith("com.azure.core")) {
+                        classType = new ClassType.Builder()
```
Comment (XiaofeiCao) [inline javagen/src/main/java/com/azure/autorest/util/SchemaUtil.java:245]: Maybe a new ClassType `PollResult`?

Comment (weidongxu-microsoft) [reply javagen/src/main/java/com/azure/autorest/util/SchemaUtil.java:245]: I am coding it a bit like an "external model", that the namespace/class is not known to codegen at coding time.

After it gets to core, I should have a predefined one.

**Pull request merged**